circleci/catapult-publish-lambda

```diff
@@ -41,38 +38,63 @@ BRANCH=$CIRCLE_BRANCH
+#    and S3_ARTIFACTS will set the S3Buckets source variable
+RUN_TYPE=$(grep type "launch/${APP_NAME}.yml" | cut -d' ' -f4)
+if [[ $RUN_TYPE == "lambda/"* ]]; then
+    AWS_REGIONS=$(echo "${RUN_TYPE}" | cut -d'/' -f2)
+    S3_ARTIFACTS=",S3Bucket=\\\"${LAMBDA_AWS_BUCKET}-${AWS_REGIONS}"
+elif [[ $RUN_TYPE == "lambda" ]]; then
+    AWS_REGIONS=${AWS_REGIONS:-"us-east-1 us-east-2 us-west-1 us-west-2"}
+    for REGION in ${AWS_REGIONS}; do
+        S3_ARTIFACTS=${S3_ARTIFACTS},"S3Buckets={${REGION}=\\\"${LAMBDA_AWS_BUCKET}-${REGION}"
+    done
+else
+    echo "Can only publish applications with run type lambda or lambda/<region>; got ${RUN_TYPE}"
+    exit 1
+fi
+
 # hack to switch from /catapult to /v2/catapult
 # TODO: should probably `ark init` things with an environment variable pointing to base url of circle-ci-integrations
 CATAPULT_URL=$(echo "${CATAPULT_URL}" | sed 's/\/catapult/\/v2\/catapult/')
 LAMBDA_AWS_S3_KEY=${APP_NAME}/${SHORT_SHA}/${APP_NAME}.zip
 
 # upload to s3
 echo "Uploading to S3..."
-# region doesn't really matter for an S3 upload, since the bucket region is fixed
-AWS_REGION=$AWS_REGION \
-          AWS_ACCESS_KEY_ID=$LAMBDA_AWS_ACCESS_KEY_ID \
-          AWS_SECRET_ACCESS_KEY=$LAMBDA_AWS_SECRET_ACCESS_KEY \
-          aws s3 cp bin/${APP_NAME}.zip s3://${LAMBDA_AWS_BUCKET}-${AWS_REGION}/${LAMBDA_AWS_S3_KEY}
-
-if [ -e swagger.yml ]; then
-    echo "Uploading swagger.yml"
-    # api gateway fails to parse on x-nullable
-    sed '/x-nullable/d' ./swagger.yml > ./swagger.lambda.yml
+for AWS_REGION in ${AWS_REGIONS}; do
+    # region doesn't really matter for an S3 upload, since the bucket region is fixed
     AWS_REGION=$AWS_REGION \
               AWS_ACCESS_KEY_ID=$LAMBDA_AWS_ACCESS_KEY_ID \
               AWS_SECRET_ACCESS_KEY=$LAMBDA_AWS_SECRET_ACCESS_KEY \
-              aws s3 cp swagger.lambda.yml s3://${LAMBDA_AWS_BUCKET}-${AWS_REGION}/${APP_NAME}/${SHORT_SHA}/swagger.lambda.yml
-fi
+              aws s3 cp bin/${APP_NAME}.zip s3://${LAMBDA_AWS_BUCKET}-${AWS_REGION}/${LAMBDA_AWS_S3_KEY}
+
+    if [ -e swagger.yml ]; then
+        echo "Uploading swagger.yml"
+        # api gateway fails to parse on x-nullable
+        sed '/x-nullable/d' ./swagger.yml > ./swagger.lambda.yml
+        AWS_REGION=$AWS_REGION \
+                  AWS_ACCESS_KEY_ID=$LAMBDA_AWS_ACCESS_KEY_ID \
+                  AWS_SECRET_ACCESS_KEY=$LAMBDA_AWS_SECRET_ACCESS_KEY \
+                  aws s3 cp swagger.lambda.yml s3://${LAMBDA_AWS_BUCKET}-${AWS_REGION}/${APP_NAME}/${SHORT_SHA}/swagger.lambda.yml
+    fi
+done;
 
 # publish the application to catapult
 echo "Publishing to catapult..."
+echo "Using data:"
+DATA="{\"username\":\"${USER}\",\"reponame\":\"${REPO}\",\"buildnum\":${BUILD_NUM},\"app\":{\"run_type\":\"${RUN_TYPE}\",\"id\":\"${APP_NAME}\",\"source\":\"github:Clever/${REPO}@${FULL_SHA}\",\"artifacts\":\"lambda:clever/${APP_NAME}@${SHORT_SHA};S3Key=\\\"${LAMBDA_AWS_S3_KEY}${S3_ARTIFACTS}\",\"branch\":\"${BRANCH}\"}}"
+echo "${DATA}"
```

Comment:
I was originally gonna remove this line, since I added it for debugging, but it seems fine to leave in? If you think it's just noise, I"m happy to axe it.